text/0000-compose.md

```diff
@@ -0,0 +1,125 @@
+* 2019-03-18
+* RFC PR: (leave this empty)
+* Fusion Issue: (leave this empty)
+
+# Summary
+
+This adds support for composing plugins via the `composePlugins` API.
+
+# Basic example
+
+```js
+import {composePlugins} from 'fusion-core';
+function composeSum(items) {
+  return items.reduce((sum, item) => {
+    return sum + item;
+  }, 0);
+}
+app.register(ComposeExampleToken, composePlugins([
+  createPlugin({
+    provides: () => 2
+  }),
+  createPlugin({
+    provides: () => 3
+  }),
+]), composeSum);
+
+// ComposeExampleToken => 5
+```
```

Comment:
The API interface looks clean and simple. Does this interface allow, each plugin having its own defined / different set of deps ?

Reply:
I think these are just example plugins, the plugins themselves could have arbitrary dependencies.

Unless I'm mistaken, the composition part is only concerned with the provided service for each plugin.

Reply:
I think these anonymous plugins could use having specific identifiers to help with error messages, & diagnostics.

Reply:
correct.